cycles-ledger/tests/tests.rs

```diff
@@ -2935,6 +2965,564 @@ fn test_icrc1_transfer_duplicate(env: &TestEnv) {
+    let total_supply_before = env.icrc1_total_supply();
+    let mut expected_blocks = env.get_all_blocks();
+
+    let args = ApproveArgs {
+        from_subaccount: account_from.subaccount,
+        spender: account_spender,
+        amount: Nat::from(amount),
+        expected_allowance: args_expected_allowance.map(Nat::from),
+        expires_at: args_expires_at,
+        fee: args_fee.map(Nat::from),
+        memo: args_memo.clone(),
+        created_at_time: args_created_at_time,
+    };
+    let block_index = env.icrc2_approve_or_trap(account_from.owner, args);
+
+    assert_eq!(
+        env.icrc1_balance_of(account_from),
+        account_from_balance_before - fee,
+    );
+    assert_eq!(
+        env.icrc1_balance_of(account_spender),
+        account_to_balance_before,
+    );
+    assert_eq!(env.icrc1_total_supply(), total_supply_before - fee,);
+    assert_eq!(
+        env.icrc2_allowance(account_from, account_spender),
+        Allowance {
+            allowance: Nat::from(1u8),
+            expires_at: args_expires_at,
+        }
+    );
+
+    let expected_new_block = Block {
+        transaction: Transaction {
+            operation: Operation::Approve {
+                from: account_from,
+                spender: account_spender,
+                amount,
+                expected_allowance: args_expected_allowance,
+                expires_at: args_expires_at,
+                fee: args_fee,
+            },
+            created_at_time: args_created_at_time,
+            memo: args_memo,
+        },
+        timestamp: env.nanos_since_epoch_u64(),
+        phash: Some(env.get_block(block_index - 1u8).hash().unwrap()),
+        effective_fee: args_fee.xor(Some(fee)),
+    };
+    expected_blocks.push(expected_new_block);
+    assert_vec_display_eq(expected_blocks, env.get_all_blocks());
+}
+
+fn test_icrc2_approve_ok_without_created_at_time(env: &TestEnv) {
+    // This function is safe to be called multiple times because
+    // the transactions it creates are not marked for deduplication.
+
+    use ShouldSetCreatedAtTime::*;
+    use ShouldSetExpectedAllowance::*;
+    use ShouldSetExpiresAt::*;
+    use ShouldSetFee::*;
+    use ShouldSetMemo::*;
+
+    for should_set_expected_allowance in [DontSetExpectedAllowance, SetExpectedAllowance] {
+        for should_set_expires_at in [DontSetExpiresAt, SetExpiresAt] {
+            for should_set_fee in [DontSetFee, SetFee] {
+                for should_set_memo in [DontSetMemo, SetMemo] {
+                    test_icrc2_approve_ok_with_params(
+                        env,
+                        DontSetCreatedAtTime,
+                        should_set_fee,
+                        should_set_memo,
+                        should_set_expected_allowance,
+                        should_set_expires_at,
+                    )
+                }
+            }
+        }
+    }
+}
+
+fn test_icrc2_approve_ok_with_created_at_time(env: &TestEnv) {
```

Comment:
Why isn't there simply another loop for the parameter `ShouldSetCreatedAtTime`  in the previous test?